portfolio/src/main/java/com/google/sps/servlets/Greeting.java

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/greeting")
+public class Greeting extends HttpServlet {
+
+  private List<String> greetings;
+
+  public void init() {
+    greetings = new ArrayList<>();
+    greetings.add("Hi! Glad you're here. ");
+    greetings.add("Hi! Thanks for coming. ");
+    greetings.add("Welcome! ");
+    greetings.add("Salutations! ");
+    greetings.add("Howdy! ");
+    greetings.add("Hi! Nice of you to visit. ");
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String greeting = greetings.get((int) (Math.random() * greetings.size()));
```

Comment:
You probably want to check that `greetings` is initialized and not empty before accessing it, and throw an exception or otherwise indicate something went wrong. While it's obvious right now that you initialize it and add items in init(), it's good practice to set up your program to fail gracefully. That way, if in the future your program changes and greetings could be empty here, you'll get your own error message which will be easy to debug.